### PR TITLE
Refactored Tezos Operations for DRY 

### PR DESCRIPTION
**Pull request by ba205**
Addressed #53, refactored TezosOperations, also created Operation interface to generalize Transactions, Delegations, and Originations for refactoring

### CODE REVIEW

File: src/tezos/TezosOperations.ts [hunk 1]
```diff
@@ -169,6 +170,32 @@ export namespace TezosOperations {
         }
     }
 
+    export async function appendRevealOperation(
```
Comment (vishakh) [inline src/tezos/TezosOperations.ts]: Please document this function.

File: src/types/Operation.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export interface Operation {
```
Comment (vishakh) [inline src/types/Operation.ts]: As this type is Tezos specific, please move the file to the Tezos package. IIRC, there is already a file containing Tezos types.

File: src/tezos/TezosOperations.ts [hunk 3]
```diff
@@ -169,6 +170,32 @@ export namespace TezosOperations {
...
+                kind: "reveal",
+                source: keyStore.publicKeyHash,
+                fee: fee.toString(),
+                counter: (Number(account.counter) + 1).toString(),
+                gas_limit: '120',
+                storage_limit: '0',
+                public_key: keyStore.publicKey
+            };
+            var operation = operations[0]
+            operation.counter = (Number(account.counter) + 2).toString()
```
Comment (vishakh) [inline src/tezos/TezosOperations.ts]: Looks like you are giving a higher counter to the reveal than the original operation. As discussed previously, the reveal should have a lower counter.

File: src/tezos/TezosOperations.ts [hunk 4]
```diff
@@ -169,6 +170,32 @@ export namespace TezosOperations {
...
+            returnedOperations = [revealOp, operation]
```
Comment (vishakh) [inline src/tezos/TezosOperations.ts]: Can we just do a regular array append?

File: src/tezos/TezosOperations.ts (hunk 3 above)
Comment (ba205) [reply src/tezos/TezosOperations.ts]: Nope! Reveal has a counter 1 less than the new operation (delegation/transaction/origination)

File: src/types/Operation.ts (hunk 2 above)
Comment (ba205) [reply src/types/Operation.ts]: Roger!

File: src/tezos/TezosOperations.ts (hunk 1 above)
Comment (ba205) [reply src/tezos/TezosOperations.ts]: Got it!

File: src/tezos/TezosOperations.ts (hunk 4 above)
Comment (ba205) [reply src/tezos/TezosOperations.ts]: Initially I wanted to do that, but the issue is either we abstract the definition of originations/delegations/transactions to the helper function, making the function an unwieldy mess, or we keep them in the original functions. The problem is the helper function is where the reveal is checked to have happened or not, so you can't just append to the original operations list. You have to edit the inputted operation's counter, to be consistent

File: src/tezos/TezosOperations.ts (hunk 3 above)
Comment (vishakh) [reply src/tezos/TezosOperations.ts]: Ah, yes!

**Pull request merged**